lib/ansible/modules/network/cloudengine/ce_stp.py

```diff
@@ -0,0 +1,969 @@
+        self.root_protection = self.module.params['root_protection'] or None
+        self.loop_protection = self.module.params['loop_protection'] or None
+
+        # state
+        self.changed = False
+        self.updates_cmd = list()
+        self.results = dict()
+        self.proposed = dict()
+        self.existing = dict()
+        self.end_state = dict()
+
+    def cli_load_config(self, commands):
+        """ Cli load configuration """
+
+        if not self.module.check_mode:
+            load_config(self.module, commands)
+
+    def cli_get_stp_config(self):
+        """ Cli get stp configuration """
+
+        regular = "| include stp"
+
+        flags = list()
+        flags.append(regular)
+        self.stp_cfg = get_config(self.module, flags)
+
+    def cli_get_interface_stp_config(self):
+        """ Cli get interface's stp configuration """
+
+        if self.interface:
+            regular = "| ignore-case section include ^interface %s$" % self.interface
+            flags = list()
+            flags.append(regular)
+            tmp_cfg = get_config(self.module, flags)
+
+            if not tmp_cfg:
+                self.module.fail_json(
+                    msg='Error: The interface %s is not exist.' % self.interface)
+
+            if "undo portswitch" in tmp_cfg:
+                self.module.fail_json(
+                    msg='Error: The interface %s is not switch mode.' % self.interface)
+
+            self.interface_stp_cfg = tmp_cfg
+
+    def check_args(self):
+        """ Check module args """
+
+        if self.cost:
+            if self.cost.isdigit():
+                if int(self.cost) < 1 or int(self.cost) > 200000000:
+                    self.module.fail_json(
+                        msg='Error: The value of cost is out of [1 - 200000000].')
+            else:
+                self.module.fail_json(
+                    msg='Error: The cost is not digit.')
+
+        if self.tc_protection_interval:
+            if self.tc_protection_interval.isdigit():
+                if int(self.tc_protection_interval) < 1 or int(self.tc_protection_interval) > 600:
+                    self.module.fail_json(
+                        msg='Error: The value of tc_protection_interval is out of [1 - 600].')
+            else:
+                self.module.fail_json(
+                    msg='Error: The tc_protection_interval is not digit.')
+
+        if self.tc_protection_threshold:
+            if self.tc_protection_threshold.isdigit():
+                if int(self.tc_protection_threshold) < 1 or int(self.tc_protection_threshold) > 255:
+                    self.module.fail_json(
+                        msg='Error: The value of tc_protection_threshold is out of [1 - 255].')
+            else:
+                self.module.fail_json(
+                    msg='Error: The tc_protection_threshold is not digit.')
+
+        if self.root_protection or self.loop_protection or self.cost:
+            if not self.interface:
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
required_together
required_one_of
require_if
```